core/ipc/rpc/camel/src/main/java/org/opennms/core/rpc/camel/CamelRpcClientFactory.java

```diff
@@ -50,19 +54,46 @@
 import org.opennms.core.rpc.api.RpcResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
 
-public class CamelRpcClientFactory implements RpcClientFactory {
+import com.google.common.util.concurrent.SimpleTimeLimiter;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.common.util.concurrent.TimeLimiter;
+
+public class CamelRpcClientFactory implements RpcClientFactory, InitializingBean, DisposableBean {
 
     private static final Logger LOG = LoggerFactory.getLogger(CamelRpcServerProcessor.class);
 
+    private static final long rpcExecTimeoutMs = Long.getLong("org.opennms.core.rpc.camel.rpcExecTimeoutMs", TimeUnit.SECONDS.toMillis(30));
```

Comment:
Is there a need for creating a new property here ?  Can't we use `"org.opennms.jms.timeout`  ?

Reply:
Good point. We can re-use that one.